backend/wmg/data/load_cube.py

```diff
@@ -13,37 +13,186 @@
+) -> str:
+    """
+    Uploads the generated cubes to S3 and writes metadata that includes
+    information about the data schema version, the snapshot id, and the location
+    of the snapshot.
+
+    TODO(prathap): This part of the docstring will be modified when work starts on ticket:
+    https://github.com/chanzuckerberg/single-cell-data-portal/issues/5166
+
+    Specifically:
+
+    1. Snapshot artifacts are stored in s3://<environment_bucket_name>/<snapshot_id>/
+       A. Note that, as of now, <snapshot_id> is a integer representing the time elapsed
+          in seconds since epoch
+       B. If the snapshot generation pipeline is run in a "remote dev environment", then the snapshot
+          is stored in s3://<environment_bucket_name>/<remote_stack_name>/<snapshot_id>/
+       C. If validation checks on the snapshot fails, the data artifact will be stored in
+          s3://<environment_bucket_name>/<snapshot_id>_validation_failed_snapshot/
+          (or s3://<environment_bucket_name>/<remote_stack_name>/<snapshot_id>_validation_failed_snapshot/)
+
+    2. The file, s3://<environment_bucket_name>/latest_snapshot_identifer, contains the latest <snapshot_id>
+       that has passed validation. If snapshot generation pipeline is run in a "remote dev environment"
+       then the file will be at path: s3://<environment_bucket_name>/<remote_stack_name>/latest_snapshot_identifer
+
+       NOTE that `latest_snapshot_identifier` will not be updated if validation checks fails on a newly generated
+       data artifact
+
+    3. After each generation of the data artifact and successful validation checks, all but the two latest
+       snapshots are deleted
+
+    Parameters
+    ----------
+    snapshot_source_path: The current source path of the cubes that need to be uploaded to s3
+    snapshot_schema_version: The schema version of the cube data
+    snapshot_id: The snapshot id is a timestamp in seconds
+    is_snapshot_validation_successful: Boolean indicating whether data validation ran sucessfully
+
+    Returns
+    -------
+    snapshot_s3_dest_path: A string that contains the location of the uploaded data
+    """
+    snapshot_id = str(snapshot_id)
+
+    # cleanup files we do not wish to upload
+    shutil.rmtree(f"{snapshot_source_path}/{INTEGRATED_ARRAY_NAME}")
+    shutil.rmtree(f"{snapshot_source_path}/{OBS_ARRAY_NAME}")
+    shutil.rmtree(f"{snapshot_source_path}/{VAR_ARRAY_NAME}")
+
+    snapshot_s3_dest_path = _get_wmg_snapshot_s3_fullpath(
+        snapshot_schema_version, snapshot_id, is_snapshot_validation_successful
+    )
+
+    sync_command = ["aws", "s3", "sync", snapshot_source_path, snapshot_s3_dest_path]
+
+    subprocess.run(sync_command)
+
+    _write_snapshot_metadata(snapshot_schema_version, snapshot_id, is_snapshot_validation_successful)
+
+    return snapshot_s3_dest_path
+
+
+###################################### PRIVATE FUNCTIONS #################################
```

Comment:
just as a general comment, i found this a bit harder to follow / review with so much of the logic wrapped in small private functions since i need to jump all over the place to figure out what's actually happening